lua/lvim/core/treesitter.lua

```diff
@@ -13,7 +13,42 @@ M.config = function()
     highlight = {
       enable = true, -- false will disable the whole extension
       additional_vim_regex_highlighting = false,
-      disable = { "latex" },
+      disable = function(lang, buf)
+        if vim.tbl_contains({ "latex" }, lang) then
+          return true
+        end
+
+        local max_filesize = 1024 * 1024
+        local ok, stats = pcall(vim.loop.fs_stat, vim.api.nvim_buf_get_name(buf))
+        if ok and stats and stats.size > max_filesize then
+          if lvim.builtin.illuminate.active then
+            vim.cmd "IlluminatePauseBuf"
+          end
+          vim.b.indent_blankline_use_treesitter = false
+          vim.b.indent_blankline_show_current_context = false
+
+          if vim.tbl_contains({ "json" }, lang) then
+            vim.cmd "NoMatchParen"
+            vim.cmd "syntax clear"
```

Comment:
```suggestion
            vim.schedule(function()
              vim.api.nvim_buf_call(buf, function()
                vim.cmd "NoMatchParen"
                vim.cmd "syntax clear"
              end)
            end)
```

Reply:
This change causes an error in notify when opening canada.json `pattern uses more memory than 'maxmempattern'`

Reply:
well I've never seen that error before :sweat_smile: 

it could be an upstream bug, so you can just keep your version instead for now.

Reply:
should I add a TODO to change this when notify is removed from rolling?

Reply:
just this without schedule works btw
```lua
vim.api.nvim_buf_call(buf, function()
  vim.cmd "NoMatchParen"
  vim.cmd "syntax clear"
end)
```